src/msg/Messenger.cc

```diff
@@ -22,10 +22,12 @@ Messenger *Messenger::create(CephContext *cct, const string &type,
 			     entity_name_t name, string lname,
 			     uint64_t nonce, uint64_t features)
 {
-  int r = -1;
-  srand(time(NULL));
-  if (type == "random")
+  static int r = -1;
```

Comment:
Yeah, but previously "r" was a regular variable which would be okay to re-use, and now you've made it a static indicator that we've seeded rand. :)